dali/pipeline/operators/util/optical_flow.h

```diff
@@ -80,11 +82,52 @@ class OpticalFlow : public Operator<Backend> {
 
 
  private:
+  /**
+   * Optical flow lazy initialization
+   */
+  void of_lazy_init(size_t width, size_t height, size_t channels, cudaStream_t stream) {
+    std::call_once(of_initialized_,
+                   [&]() {
+                       optical_flow_.reset(
+                               new optical_flow::OpticalFlowTuring(of_params_, width, height,
+                                                                   channels, stream));
+                   });
+  }
+
+
+  /**
+   * Use input TensorList to extract calculation params
+   * Currently only FNHWC layout is supported
+   */
+  void ExtractParams(const TensorList<Backend> &tl) {
+    auto shape = tl.shape();
+    nsequences_ = shape.size();
+    frames_height_ = shape[0][1];
+    frames_width_ = shape[0][2];
+    depth_ = shape[0][3];
+    sequence_sizes_.reserve(nsequences_);
+    for (int i = 0; i < nsequences_; i++) {
+      sequence_sizes_[i] = shape[i][0];
+    }
+
+    DALI_ENFORCE([&]() -> bool {
```

Comment:
why not:
```
for (...) 
  DALI_ENFORCE(seq[1] == frames_height_ && seq[2] == frames_width_ && seq[3] == depth_);
```

Reply:
So that it's clear, that this loop is only for DALI_ENFORCE. That's how I'd do it with `assert` also ;)